components/sup/src/manager/service/supervisor.rs

```diff
@@ -101,9 +109,7 @@ impl Supervisor {
             self.change_state(ProcessState::Up);
         } else {
             self.change_state(ProcessState::Down);
-            if let Some(ref pid_file) = self.pid_file {
-                Self::cleanup_pidfile(pid_file.clone());
-            }
+            Self::cleanup_pidfile(self.pid_file.clone());
```

Comment:
Very minor - we could change the signature of `cleanup_pidfile` to `fn cleanup_pidfile(pid_file: impl AsRef<Path>)` and avoid the `clone`. `Self::cleanup_pidfile(&self.pid_file);`

Reply:
@davidMcneil Good catch; addressed in https://github.com/habitat-sh/habitat/pull/7526/commits/41b26e2e738f031a477fb940fc3e43474c1c7c85